spec/requests/orders_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+RSpec.describe "Orders", type: :request do
+
+  describe "GET /index" do
+    it "renders the index view" do
+      FactoryBot.create_list(:order, 10)
+      get orders_path
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe "GET /orders/[order]" do
+    it "renders the :show template" do
+      order = FactoryBot.create(:order)
+      get order_path(id: order.id)
+      expect(response).to render_template(:show)
+    end
+    it "redirects to the index path if the order id is invalid" do
+      get order_path(id: 5000) #an ID that doesn't exist
+      expect(response).to redirect_to orders_path
+    end
+  end
+
+  describe "GET /orders/new" do
+    it "renders the :new template" do
+      get new_order_path
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "GET /orders/[order]/edit" do
+    it "renders the :edit template" do
+      order = FactoryBot.create(:order)
+      get edit_order_path(id: order.id)
+      expect(response).to render_template(:edit)
+    end
+    it "redirects to the index path if the order id is invalid" do
+      get edit_order_path(id: 5000) #an ID that doesn't exist
+      expect(response).to redirect_to orders_path
+    end
+  end
+
+  describe "POST /orders with valid data" do
+    it "saves a new entry and redirects to the show path for the entry" do
+      customer = FactoryBot.create(:customer)
+      order_attributes = FactoryBot.attributes_for(:order, customer_id: customer.id)
+      expect { post orders_path, params: {order: order_attributes}
+    }.to change(Order, :count)
+      expect(response).to redirect_to order_path(id: Order.last.id)
+    end
+  end
+
+  describe "POST /orders with invalid data" do
+    it "does not save a new entry or redirect" do
+      order_attributes = FactoryBot.attributes_for(:order)
+      order_attributes.delete(:product_id)
+      expect { post orders_path, params: {order: order_attributes}
+    }.to_not change(Order, :count)
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "PUT /orders/[order] with valid data" do
+    it "updates an entry and redirects to the show path for the order" do
+      order = FactoryBot.create(:order)
+      put "/orders/#{order.id}", params: {order: {product_count: 50}}
```

Comment:
It's generally better to use the URL helper when you can.

```suggestion
      put order_path(order.id), params: {order: {product_count: 50}}
```